### PR TITLE
8327242: Document supported CLDR versions in the javadoc

### PR DESCRIPTION
**Pull request by naotoj**
Adding a table that maps JDK versions and corresponding CLDR releases as an implNote. A draft CSR has also been created.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change requires CSR request [JDK-8327259](https://bugs.openjdk.org/browse/JDK-8327259) to be approved

### Issues
 * [JDK-8327242](https://bugs.openjdk.org/browse/JDK-8327242): Document supported CLDR versions in the javadoc (**Bug** - P4)
 * [JDK-8327259](https://bugs.openjdk.org/browse/JDK-8327259): Document supported CLDR versions in the javadoc (**CSR**)

### Reviewers
 * [Joe Wang](https://openjdk.org/census#joehw) (@JoeWang-Java - **Reviewer**) ⚠️ Review applies to [98b40f95](https://git.openjdk.org/jdk/pull/18243/files/98b40f956fb4f0953b69825a8bdc939bd1843747)
 * [Iris Clark](https://openjdk.org/census#iris) (@irisclark - **Reviewer**)
 * [Justin Lu](https://openjdk.org/census#jlu) (@justin-curtis-lu - Committer) ⚠️ Review applies to [98b40f95](https://git.openjdk.org/jdk/pull/18243/files/98b40f956fb4f0953b69825a8bdc939bd1843747)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/18243/head:pull/18243` \
`$ git checkout pull/18243`

Update a local copy of the PR: \
`$ git checkout pull/18243` \
`$ git pull https://git.openjdk.org/jdk.git pull/18243/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 18243`

View PR using the GUI difftool: \
`$ git pr show -t 18243`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/18243.diff">https://git.openjdk.org/jdk/pull/18243.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/18243#issuecomment-1992078921)

### CODE REVIEW

**Review by JoeWang-Java — APPROVED** [review]
Irrelevant to this doc change, I happen to notice some of the jdk`<version>`-suported-locales docs might have not been updated. For example, https://www.oracle.com/java/technologies/javase/jdk20-suported-locales.html had "CLDR release 39" rather than 42 as stated here. Similarly jdk19-suported-locales.html -> 39 instead of 41, 18 was correctly -> 39, but 17 -> 35.1, and so on.

File: src/java.base/share/classes/java/util/spi/LocaleServiceProvider.java [hunk 1]
```diff
@@ -144,6 +145,51 @@
  * require implementations of the locale sensitive services must explicitly specify
  * "SPI" in order for the Java runtime to load them from the classpath.
  *
+ * @implNote The JDK uses locale data from the Unicode Common Locale Data Repository
+ * (CLDR) to implement locale-sensitive APIs in the {@code java.util} and
+ * {@code java.text} packages. This locale data makes the set of locales
+ * supported by the Java runtime environment. The following table lists the
+ * version of CLDR used in each JDK release.
+ * <table class="striped">
+ * <caption style="display:none">Shows JDK releases and supported CLDR versions</caption>
```
[the remark this answers is not in the document]
Comment (naotoj) [reply src/java.base/share/classes/java/util/spi/LocaleServiceProvider.java]: Good point. Removed.

File: src/java.base/share/classes/java/util/spi/LocaleServiceProvider.java [hunk 2]
```diff
@@ -144,6 +145,51 @@
  * require implementations of the locale sensitive services must explicitly specify
  * "SPI" in order for the Java runtime to load them from the classpath.
  *
+ * @implNote The JDK uses locale data from the Unicode Common Locale Data Repository
```
[the remark this answers is not in the document]
Comment (naotoj) [reply src/java.base/share/classes/java/util/spi/LocaleServiceProvider.java]: Used the full name in both places, as they are the locations that refer to the Unicode's locale data. Other "CLDR" represents the JDK's CLDR locale data provider.

**Comment by naotoj** [comment]
> Irrelevant to this doc change, I happen to notice some of the jdk`<version>`-suported-locales docs might have not been updated. For example, https://www.oracle.com/java/technologies/javase/jdk20-suported-locales.html had "CLDR release 39" rather than 42 as stated here. Similarly jdk19-suported-locales.html -> 39 instead of 41, 18 was correctly -> 39, but 17 -> 35.1, and so on.

Thanks for pointing it out. Will let the doc people know. It is at least good that the latest doc for JDK21 is correct.

**Review by irisclark — APPROVED** [review]

**Review by justin-curtis-lu — APPROVED** [review]
LGTM

**Review by JoeWang-Java — APPROVED** [review]

File: src/java.base/share/classes/java/util/spi/LocaleServiceProvider.java [hunk 3]
```diff
@@ -139,11 +140,57 @@
  * the locale sensitive services in the SPI providers are looked up first. If the
  * desired locale sensitive service is not available, then the runtime looks for CLDR.
  * <p>
- * The default value for looking up the preferred locale providers is "CLDR",
+ * The default value for looking up the preferred locale data providers is "CLDR",
  * so specifying "CLDR" is identical to the default behavior. Applications which
```
[the remark this answers is not in the document]
Comment (naotoj) [reply src/java.base/share/classes/java/util/spi/LocaleServiceProvider.java:144]: Thanks. Modified.

**Review by irisclark — APPROVED** [review]

**Comment by naotoj** [comment]
>  Will let the doc people know. It is at least good that the latest doc for JDK21 is correct.

Now the previous releases' supported locales docs have been updated.

**Pull request closed without merging**